src/vsr/journal.zig

```diff
@@ -1237,32 +1289,30 @@ pub fn Journal(comptime Replica: type, comptime Storage: type) type {
             }
 
             const checkpoint_index = self.slot_for_op(replica.op_checkpoint).index;
-            if (checkpoint_index == torn_slot.index) {
-                // The checkpoint and the torn op are in the same slot.
-                assert(cases[checkpoint_index].decision(replica.replica_count) == .vsr);
-                assert(slot_count > 1);
-                assert(op_max >= replica.op_checkpoint);
-                assert(torn_op == op_max + 1);
-                assert(torn_op > replica.op_checkpoint);
-                return null;
-            }
-
             const known_range = SlotRange{
                 .head = Slot{ .index = checkpoint_index },
                 .tail = torn_slot,
             };
 
             // We must be certain that the torn prepare really was being appended to the WAL.
-            // Return if any faults do not lie between the checkpoint and the torn prepare, such as:
+            // Return null if any faults do not lie between the checkpoint and the torn prepare,
+            // such as:
```

Comment:
Would be great just to explain "why" a little more here for a first time reader.

i.e. "and the torn prepare, such as: ... Since then we cannot be certain if the prepare was torn or corrupted.